lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_extract_keys_from_strings_files.rb

```diff
@@ -0,0 +1,131 @@
+      # @return [Hash<String, Array<String>>] The hash listing the keys to extract for each target file
+      #
+      def self.keys_list_per_target_file(original_files)
+        original_files.map do |original_file|
+          keys = Fastlane::Helper::Ios::L10nHelper.read_strings_file_as_hash(path: original_file).keys
+          [original_file, keys]
+        end.to_h
+      rescue StandardError => e
+        UI.user_error!("Failed to read the keys to extract from originals file: #{e.message}")
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Extracts a subset of keys from a `.strings` file into a separate `.strings` file'
+      end
+
+      def self.details
+        <<~DETAILS
+          Extracts a subset of keys from a `.strings` file into a separate `.strings` file, for each `*.lproj` subdirectory.
+
+          This is especially useful to extract keys for `InfoPlist.strings` or `<SomeIntentDefinitionFile>.strings`
+          from the `Localizable.strings` file, for each locale.
+
+          Since we typically merge all `*.strings` original files (e.g. `en.lproj/Localizable.strings` + `en.lproj/InfoPlist.strings`)
+          via `ios_merge_strings_file` before sending the originals to translations, this is why we then need to extract
+          the relevant keys and translations back into `InfoPlist.strings` after we pull those translations back from GlotPress
+          (`ios_download_strings_files_from_glotpress`) using this `ios_extract_keys_from_strings_files` action.
+        DETAILS
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :source_parent_dir,
+                                       env_name: 'FL_IOS_EXTRACT_KEYS_FROM_STRINGS_FILES_SOURCE_PARENT_DIR',
+                                       description: 'The parent directory containing all the `*.lproj` subdirectories in which the `.strings` files reside',
+                                       type: String,
+                                       verify_block: proc do |value|
+                                         UI.user_error!("`source_parent_dir` should be a path to an existing directory, but found #{value}.") unless File.directory?(value)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :source_tablename,
+                                       env_name: 'FL_IOS_EXTRACT_KEYS_FROM_STRINGS_FILES_SOURCE_TABLENAME',
+                                       description: 'The basename of the `.strings` file (without the extension) to extract the keys and translations from',
+                                       type: String,
+                                       default_value: 'Localizable'),
+          FastlaneCore::ConfigItem.new(key: :target_original_files,
```

Comment:
Ok after the week-end I think I'm ok with the action name (`extract_keys_from_strings_files`) after all… but now my next naming bikeshed is about this one.

The idea is to have a list of `<path>/<base_locale>/<target_table>.strings` paths, which, for each target path, will convey the information of { the relative path of the parent dir containing the `.lproj` for the _target_ files (which might not be the same as the source one) + the base locale + the target table name } all in a single string for each item of the array (as opposed to having something like an Array of Hashes).

_For example when we'll want to extract the keys from `Localizable.strings` (downloaded from GlotPress) into both `InfoPlist.strings` and `Sites.strings`, because those are for different Xcode targets and in different folders (`WordPress/Resources` vs `WordPress/WordPressIntents`, respectively), we need, for each target table, to provide both the path to the parent dir containing the `.lproj`, and the basename of the table._

---

This approach is a nicer and more flexible API than the previous one I drafted imho… but I'm unsure if the name of the parameter (`target_original_files:`) is the base name we could come up then. I like the `target_*` prefix as it makes for a nice counterpart to `source_*` for the other params, but not sure if the name is clear enough. Maybe `target_files_originals` would be better, maybe something else… 🤷